valispace/__init__.py

```diff
@@ -110,13 +111,37 @@ def get_vali(self, id=None, name=None, workspace_id=None, project_id=None, paren
 				print("VALISPACE-ERROR: Vali id must be an integer")
 				return
 
+		# Check if no argument was passed
+		if id is None and name is None:
+			print("VALISPACE-ERROR: 1 argument expected (name or id)")
+			return
+
+		url = self.valispace_login['url'] + "vali/?"
+		if id:
+			url += "id=" + str(id)
+		elif name:
+			url += "name=" + str(name)
+
+		# Access API
+		headers = self.valispace_login['options']['Headers']
+
+		response = requests.get(url, headers=headers)
+
+		return response.json()
+
+	def filter_vali(self, workspace_id=None, workspace_name=None, project_id=None, project_name=None, parent_id=None,
+																	parent_name=None, tag_id=None, tag_name=None, vali_marked_as_impacted=None):
+		""" Returns JSON with all the Valis that mach the input arguments.
+		Inputs are integers for IDs or vali_marked_as_impacted strings, and strings for names.
+		Use the component 'unique_name' (not the 'name') in the parent_name argument."""
```

Comment:
No normal user knows what the 'unique name' means, it's something internal.

Reply:
Ok, so maybe I change it back to "name" but I put a note about the valis and components need to have the tree path in the name in order for the names to be unique?

Reply:
Yes that sounds like a good idea.